resources/assets/js/components/Modal.tsx

```diff
@@ -1,189 +1,168 @@
+
+    return (): void => document.removeEventListener("keydown", keyListener);
+  }, [children]);
+
+  return createPortal(
+    <div
+      aria-describedby={`${id}-description`}
+      aria-hidden={!visible}
+      aria-labelledby={`${id}-title`}
+      data-c-dialog={visible ? overflow : ""}
+      data-c-dialog-id={id}
+      data-c-padding="top(double) bottom(double)"
+      role="dialog"
+      ref={modalRef}
+    >
+      <div data-c-background="white(100)" data-c-radius="rounded">
+        <modalContext.Provider
+          value={{ id, parentElement, visible, onModalConfirm, onModalCancel }}
         >
-          <div data-c-background="white(100)" data-c-radius="rounded">
-            <div
-              data-c-padding="normal"
-              data-c-border="bottom(thin, solid, black)"
-            >
-              <h5 data-c-font-size="h4" id={`${id}-title`}>
-                {title}
-              </h5>
-
-              {subtitle && (
-                <span data-c-font-size="h5" data-c-margin="top(half)">
-                  {subtitle}
-                </span>
-              )}
-
-              <button
-                data-c-dialog-action="close"
-                data-c-dialog-id={id}
-                type="button"
-                onClick={this.handleClose}
-              >
-                <i className="material-icons">{closeText}</i>
-              </button>
-            </div>
-
-            <div
-              data-c-border="bottom(thin, solid, black)"
-              data-c-padding="normal"
-            >
-              <div id={`${id}-description`}>{children}</div>
-            </div>
-
-            <div data-c-padding="normal">
-              <div data-c-grid="gutter middle">
-                <div data-c-grid-item="base(1of2)">
-                  <button
-                    data-c-button="outline(slow)"
-                    data-c-radius="rounded"
-                    data-c-dialog-action="close"
-                    data-c-dialog-id={id}
-                    type="button"
-                    onClick={this.handleClose}
-                  >
-                    {cancelText}
-                  </button>
-                </div>
-                <div
-                  data-c-grid-item="base(1of2)"
-                  data-c-alignment="base(right)"
-                >
-                  <button
-                    data-c-button="solid(go)"
-                    data-c-radius="rounded"
-                    data-c-dialog-action="close"
-                    data-c-dialog-id={id}
-                    type="button"
-                    onClick={handleConfirm}
-                  >
-                    {confirmText}
-                  </button>
-                </div>
-              </div>
-            </div>
-          </div>
-        </div>
-      </>
-    );
-  }
+          {children}
+        </modalContext.Provider>
+      </div>
+    </div>,
+    parentElement && parentElement.length > 0
+      ? document.querySelector(parentElement)
```

Comment:
It makes me nervous that we pass in a querySelector string into the modal, and hope it works. I'd rather pass in the dom node directly.